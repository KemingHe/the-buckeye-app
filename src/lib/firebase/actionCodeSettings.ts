"use client";

import type { ActionCodeSettings } from "firebase/auth";

import { AUTH_VERIFY_ROUTE } from "@constants/routeConstants";

export const FIREBASE_AUTH_LOCAL_STORAGE_KEY: string = "firebaseAuthEmail";

// Firebase client auth (web app only) action code settings.
export const actionCodeSettings: ActionCodeSettings = {
  url: `${window.location.origin}${AUTH_VERIFY_ROUTE}`,
  handleCodeInApp: true
};
