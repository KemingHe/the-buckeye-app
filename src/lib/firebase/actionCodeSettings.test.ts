import { describe, expect, it } from "vitest";

import isNonEmptyString from "@utils/isNonEmptyString";

import * as settings from "./actionCodeSettings";

describe("actionCodeSettings", () => {
  it("exports the correct firebase client action code settings object", () => {
    const actionCodeSettings = settings.actionCodeSettings;
    expect(isNonEmptyString(actionCodeSettings.url)).toBe(true);
    expect(actionCodeSettings.handleCodeInApp).toBe(true);
  });

  it("exports the correct firebase auth local storage key", () => {
    expect(settings.FIREBASE_AUTH_LOCAL_STORAGE_KEY).toMatchSnapshot();
  });
});
