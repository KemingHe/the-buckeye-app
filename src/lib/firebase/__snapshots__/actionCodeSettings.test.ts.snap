// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`actionCodeSettings > exports the correct firebase auth local storage key 1`] = `"firebaseAuthEmail"`;
